build.gradle

```diff
@@ -132,9 +136,17 @@ subprojects {
         testRuntimeOnly "org.slf4j:slf4j-log4j12:$slf4jVersion"
     }
 
-    tasks.named('test') {
+    test {
```

Comment:
Please read gradle guidlines https://docs.gradle.org/current/userguide/task_configuration_avoidance.html#sec:task_configuration_avoidance_pitfalls
This should actually be applied everywhere and I was trying to do it before but maybe I missed some places.

Reply:
Thanks, @AnatolyPopov! Wasn't aware of this. Gradle can be very tricky sometimes.

Though, this is probably referring to changes from #328. Applied the fixes there. 

I have removed the dependency on that PR as changes here can be applied individually. Have a look